lib/rspec/mocks.rb

```diff
@@ -101,6 +101,23 @@ def self.with_temporary_scope
       end
     end
 
+    @@excluded_subclasses = []
+
+    def self.excluded_subclasses
```

Comment:
Is it necessary to getobj?
There’s a potential race condition, and the weakref itself should quack just like the object itself (including the === probably to allow to remove from the list.
https://stackoverflow.com/questions/69185508/ruby-weakref-has-implicit-race-condition

Reply:
Yes, otherwise the substraction wasn't working.

I changed to rescue RefError.